lib/src/object_store.rs

```diff
@@ -1,15 +1,29 @@
 use {
-    crate::wiggle_abi::types::FastlyStatus,
+    crate::wiggle_abi::types::{FastlyStatus, KvError, KvInsertMode},
+    base64::prelude::*,
+    serde::Serialize,
     std::{
         collections::BTreeMap,
         sync::{Arc, RwLock},
+        time::SystemTime,
     },
 };
 
+#[derive(Debug, Clone)]
+pub struct ObjectValue {
+    pub body: Vec<u8>,
+    // these two replace an Option<String> so we can
+    // derive Copy
+    pub metadata: Vec<u8>,
+    pub metadata_len: usize,
+    pub generation: u32,
+    pub expiration: Option<SystemTime>,
+}
+
 #[derive(Clone, Debug, Default)]
 pub struct ObjectStores {
     #[allow(clippy::type_complexity)]
-    stores: Arc<RwLock<BTreeMap<ObjectStoreKey, BTreeMap<ObjectKey, Vec<u8>>>>>,
+    stores: Arc<RwLock<BTreeMap<ObjectStoreKey, BTreeMap<ObjectKey, ObjectValue>>>>,
 }
 
 impl ObjectStores {
```

Comment:
Can you add some units tests to the end of this file that verifies the different `ObjectStores` methods? This seems like the easiest way to verify the new list functions here until the SDK changes get published.

Reply:
Absolutely.

FYI, I do do manual testing with the mock app/integration, also deployment in the lab, and then local tests with viceroy. Not saying that that checks the same boxes, but just so you know :P

Reply:
Ok tests added!